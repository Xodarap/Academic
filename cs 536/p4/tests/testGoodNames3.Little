int g;
double g1;

void f() {}
int f1() {}

void multParams1(int x, double y, int z) {}
void multParams2(int x, double y, int z) {}
void multParams3(int x, double y) {}

/* function with decls of vars in all stmtlists */
void locals() {
  int x;
  double y;
  if (0) {
    int y;
    double x;
  }
  if (0) {
  } else {
    int y;
    double x;
  }
  while (1) {
    int y;
    double x;
  }
}

/* function with uses of vars in all stmts */
int uses() {
  int x;
  double d;
  g1 = g;
  ++x;
  --x;
  x++;
  x--;
  scanf("%d", &x);
  scanf("%f", &d);
  printf("%d", x);
  printf("%f", d);
  if (x) {
    int y;
    x = y;
  }
  if (x) {
  } else {
    int y;
    x = y;
  }
  while (x) {
    int y;
    x = y;
  }
  return x;
  f();
  multParams1(x, g1, 22);
}

/* function with uses of declared vars in all exprs */
void declaredInExps() {
  int x;
  int y;
  int z;
  x = y = z;
  x = y + z;
  x = y - z;
  x = y * z;
  x = y / z;
  x = ! z;
  x = y && z;
  x = y || z;
  x = y == z;
  x = y != z;
  x = y < z;
  x = y > z;
  x = y <= z;
  x = y >= z;
  x = -y;
  x = y++;
  x = y--;
}