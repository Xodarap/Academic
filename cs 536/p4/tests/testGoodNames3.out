int g;
double g1;

void f() {
}

int f1() {
}

void multParams1(int x, double y, int z) {
}

void multParams2(int x, double y, int z) {
}

void multParams3(int x, double y) {
}

void locals() {
  int x;
  double y;
  if (0) {
    int y;
    double x;
  }
  if (0) {
  }
  else {
    int y;
    double x;
  }
  while (1) {
    int y;
    double x;
  }
}

int uses() {
  int x;
  double d;
  g1(double) = g(int);
  ++x(int);
  --x(int);
  x(int)++;
  x(int)--;
  scanf("%d", &x(int));
  scanf("%f", &d(double));
  printf("%d", x(int));
  printf("%f", d(double));
  if (x(int)) {
    int y;
    x(int) = y(int);
  }
  if (x(int)) {
  }
  else {
    int y;
    x(int) = y(int);
  }
  while (x(int)) {
    int y;
    x(int) = y(int);
  }
  return x(int);
  f(->void)();
  multParams1(int,double,int->void)(x(int), g1(double), 22);
}

void declaredInExps() {
  int x;
  int y;
  int z;
  x(int) = (y(int) = z(int));
  x(int) = (y(int)+z(int));
  x(int) = (y(int)-z(int));
  x(int) = (y(int)*z(int));
  x(int) = (y(int)/z(int));
  x(int) = (!z(int));
  x(int) = (y(int)&&z(int));
  x(int) = (y(int)||z(int));
  x(int) = (y(int)==z(int));
  x(int) = (y(int)!=z(int));
  x(int) = (y(int)<z(int));
  x(int) = (y(int)>z(int));
  x(int) = (y(int)<=z(int));
  x(int) = (y(int)>=z(int));
  x(int) = (-y(int));
  x(int) = (y(int)++);
  x(int) = (y(int)--);
}
